src/main/kotlin/de/gleex/pltcmd/ui/GeneratingView.kt

```diff
@@ -0,0 +1,64 @@
+package de.gleex.pltcmd.ui
+
+import de.gleex.pltcmd.model.mapgenerators.ui.IncompleteMapBlock
+import de.gleex.pltcmd.model.mapgenerators.ui.IncompleteMapGameArea
+import de.gleex.pltcmd.options.UiOptions
+import org.hexworks.zircon.api.Components
+import org.hexworks.zircon.api.GameComponents
+import org.hexworks.zircon.api.component.ComponentAlignment
+import org.hexworks.zircon.api.component.Label
+import org.hexworks.zircon.api.component.Panel
+import org.hexworks.zircon.api.component.ProgressBar
+import org.hexworks.zircon.api.data.Size
+import org.hexworks.zircon.api.data.Tile
+import org.hexworks.zircon.api.grid.TileGrid
+import org.hexworks.zircon.api.view.base.BaseView
+
+/** Displays the progress of world generation. A miniature of the world is shown together with a progress bar. */
+class GeneratingView(tileGrid: TileGrid) : BaseView(theme = UiOptions.THEME, tileGrid = tileGrid) {
+
+    private val progressBar = createProgressBar()
+    private val headerHeight = 1
+    private val usedLines = progressBar.height + headerHeight
+    val progressProperty = progressBar.progressProperty
+    val incompleteWorld = IncompleteMapGameArea(Size.create(screen.width, screen.height - usedLines))
+
+    override fun onDock() {
+        val header = createHeader()
+        val mainPart = createMainPart()
+
+        screen.addComponents(header, mainPart, progressBar)
+    }
+
+    private fun createHeader(): Label {
+        return Components.label()
```

Comment:
```suggestion
    private fun createHeader(): Header {
        return Components.header()
```
1. It does what it says it would ;)
2. It is much more readable. When I started it up the first couple of times I didn't even notice there was text above the map.